monitor/sth_fetcher.go

```diff
@@ -105,28 +187,133 @@ func (f *sthFetcher) stop() {
 // since the STH's timestamp is published to the `sth_age` metric. If an error
 // occurs the `sth_failures` metric will be incremented. If the operation
 // succeeds then the `sth_timestamp` gauge will be updated to the returned STH's
-// timestamp.
+// timestamp. The newly observerd STH will be stored as `f.prevSTH`. If
+// `f.prevSTH` is not nil, then `observeSTH` will asynchronously validate
+// consistency between `f.prevSTH` and the newly observed STH.
 func (f *sthFetcher) observeSTH() {
 	labels := prometheus.Labels{"uri": f.logURI}
-	f.logger.Printf("Fetching STH for %q\n", f.logURI)
+	f.log("Fetching STH")
 
 	start := f.clk.Now()
 	ctx, cancel := context.WithTimeout(context.Background(), f.sthTimeout)
 	defer cancel()
-	sth, err := f.client.GetSTH(ctx)
+	newSTH, err := f.client.GetSTH(ctx)
 	elapsed := f.clk.Since(start)
 	f.stats.sthLatency.With(labels).Observe(elapsed.Seconds())
 
 	if err != nil {
-		f.logger.Printf("!! Error fetching STH: %s\n", err.Error())
+		f.logErrorf("failed to fetch STH: %q : %#v", err.Error(), err)
```

Comment:
Can you give an example in the comments of what `err.Error()` vs `%#v` / `err` look like, to explain why this error log needs both?

Reply:
The `ct.Client.GetSTH` function can end up returning a `client.RespError` instance: https://github.com/google/certificate-transparency-go/blob/3b560b3bad9d25eb9dbe2c92b8726efecb65a4c3/client/logclient.go#L174-L176

The `RespError` interface has information like the HTTP Status code & response body that I found useful in debugging, but its `Error()` method only prints the inner error's error message:
https://github.com/google/certificate-transparency-go/blob/3b560b3bad9d25eb9dbe2c92b8726efecb65a4c3/client/logclient.go#L59-L70

Including `%#v` meant that the RespError details are there as well.

Reply:
Left a brief comment indicating the above in 9c7d759